tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -1035,3 +1037,73 @@ def process(self, resources, event=None):
         parent_resources_ids = [p['id'] for p in parent_resources]
         parent_key = self.manager.resource_type.parent_key
         return [r for r in resources if r[parent_key] in parent_resources_ids]
+
+
+class AzureAdvisorFilter(RelatedResourceFilter):
+    """
+    Filter resources by Azure Advisor Recommendations
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: disks-with-cost-recommendations
+            resource: azure.disk
+            filters:
+              - type: advisor-recommendation
+                key: '[].properties.category'
+                value: Cost
+                value_type: swap
+                op: in
+
+    """
+
+    RelatedResource = "c7n_azure.resources.advisor.AdvisorRecommendation"
+    RelatedIdsExpression = "id"
+    AnnotationKey = "AdvisorRecommendation"
+
+    _recommendation_map = {}
+
+    schema = type_schema(
+        "advisor-recommendation",
+        rinherit=RelatedResourceFilter.schema,
+    )
+
+    def _add_annotations(self, related_ids, resource):
+        resource[f"c7n:{self.AnnotationKey}"] = [
+            r["id"] for r in
```

Comment:
It seems useful to add the full recommendation here rather than just the ID, so folks can see the recommendations and report on them through `resources.json`

Reply:
i did consider that, my concern was the increase in data stored on the annotation, at one point i had a `detailed_annotation` flag in the schema to allow you to do a full print out of the recommendations